environments/common/src/main/java/org/jboss/weld/environment/deployment/discovery/DiscoveryStrategyFactory.java

```diff
@@ -42,20 +47,33 @@ private DiscoveryStrategyFactory() {
      */
     public static DiscoveryStrategy create(ResourceLoader resourceLoader, Bootstrap bootstrap,
         Set<Class<? extends Annotation>> initialBeanDefiningAnnotations, boolean jandexStrategyDisabled) {
-        if (Jandex.isJandexAvailable(resourceLoader)) {
+        DiscoveryStrategy returnValue = null;
+        final Iterator<Metadata<DiscoveryStrategy>> iterator = ServiceLoader.load(DiscoveryStrategy.class, resourceLoader).iterator();
+        if (iterator != null && iterator.hasNext()) {
+            final DiscoveryStrategy candidate = iterator.next().getValue();
```

Comment:
It could be useful to log a DEBUG message that a service provider is used...?

Reply:
I like this idea, we should use the `org.jboss.weld.environment.logging.CommonLogger` for it.

Reply:
OK; I'll invent a message and add it to `CommonLogger` and use it here.